integration/utils/contLoadGenerator.go

```diff
@@ -380,6 +455,125 @@ func (lg *ContLoadGenerator) sendAddKeyTx(workerID int) {
 	<-ch
 }
 
+func (lg *ContLoadGenerator) addKeysToProposerAccount(proposerPayerAccount *flowAccount) error {
+	if proposerPayerAccount == nil {
+		return errors.New("proposerPayerAccount is nil")
+	}
+
+	addKeysToPayerTx, err := lg.createAddKeyTx(*lg.accounts[0].address, lg.constExecParam.PayerKeyCount)
+	if err != nil {
+		lg.log.Error().Msg("failed to create add-key transaction for const-exec")
+		return err
+	}
+	addKeysToPayerTx.SetReferenceBlockID(lg.follower.BlockID()).
+		SetProposalKey(*lg.accounts[0].address, 0, lg.accounts[0].seqNumber).
+		SetPayer(*lg.accounts[0].address)
+
+	lg.log.Info().Msg("signing the add-key transaction for const-exec")
+	err = lg.accounts[0].signTx(addKeysToPayerTx, 0)
+	if err != nil {
+		lg.log.Error().Err(err).Msg("error signing the add-key transaction for const-exec")
+		return err
+	}
+
+	lg.log.Info().Msg("issuing the add-key transaction for const-exec")
+	ch, err := lg.sendTx(0, addKeysToPayerTx)
+	if err != nil {
+		return err
+	}
+	<-ch
+
+	lg.log.Info().Msg("the add-key transaction for const-exec is done")
+	return nil
+}
+
+func (lg *ContLoadGenerator) getTxSizeWithoutCommnent() uint {
+	emptyTxBody := flow.TransactionBody{}
+	emptyTxBodySize := unsafe.Sizeof(emptyTxBody)
+	authorizerListStr := generateAuthAccountParamList(lg.constExecParam.AuthAccountNum)
+	oneSignatureLen := uint(flow.AddressLength) + flowCrypto.SignatureLenECDSAP256 + 8 /* keyId is uint64 */
+
+	return lg.constExecParam.ArgSizeInByte +
```

Comment:
is there a better way to get the size estimate?  this seems very brittle.

Reply:
Janez and I discussed about this. Basically we want to know the size of binary data that takes actual network resources. Our goal is to reach the network saturation point and then collect some metrics. I will leave it to @janezpodhostnik for more info

Reply:
For the required measurements the actual absolute size is not that important. Let me explain what I mean:

If we write the "size" of the transaction as $T_0 + \Delta T_x$ the minimum size of a transaction with no parameters, comments, authorisers,... + the variable part. The quantity we will measure is the max TPS (the saturation point)  of $T_x$ $S_x$ , and what we really interested in  is $K$ where $C + K (T_0 + \Delta T_x) = S_x$. Which we can get with:

$$
\displaylines{ C + K (T_0 + \Delta T_1) &=& S_1 \\
C + K (T_0 + \Delta T_2) &=& S_2 \\
(\Delta T_1-\Delta T_2) K &=& S_1-S_2}
$$

In general the "size" is a vector of all the different parameters, and we have more than two points, but the point is the accuracy of the constant size of the transaction is not important as we are only interested in the relative size.

However I agree this might be brittle and confusing, so we could instead just not count the constant parts, and only count the dynamic parts:
```
                lg.constExecParam.ArgSizeInByte +
                generateAuthAccountParamList(lg.constExecParam.AuthAccountNum) +
   		lg.constExecParam.AuthAccountNum*oneSignatureLen +
		lg.constExecParam.PayerKeyCount*oneSignatureLen
```

This will make the comparison `lg.constExecParam.MaxTxSizeInByte < txSizeWithoutComment` inaccurate, but if the constant parts are not that big we can just rewrite it to:  `lg.constExecParam.MaxTxSizeInByte < txRelativeSizeWithoutComment + TXConstantSizeEstimate`.

Reply:
can we just marshal the payload without sending it, and get the size from that?

Reply:
addressed - now tx size is calculated by RLP encoding which is one simple API call away: `tx.Encode()`. Using protobuf encoding to calculate the size seems to be a little more complex and adds extra assumption that we will be using protobuf for data transfer.